partiql-ast/src/main/java/org/partiql/ast/v1/type/TypeVarchar.java

```diff
@@ -0,0 +1,32 @@
+package org.partiql.ast.v1.type;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.partiql.ast.v1.AstNode;
+import org.partiql.ast.v1.AstVisitor;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * TODO docs, equals, hashcode
+ */
+public class TypeVarchar extends Type {
```

Comment:
Just pointing out the `VARCHAR(l)` and `CHARACTER VARYING(l)` are represented by `TypeVarchar` -- but `INT4`, `INTEGER`, and `INT` are represented by two different types: `TypeInt` and `TypeInt4`.